src/services/download/s3.rs

```diff
@@ -11,13 +11,24 @@ use futures::TryStreamExt;
 use parking_lot::Mutex;
 use rusoto_s3::S3;
 
-use super::locations::{join_prefix_location, SourceLocation};
+use super::locations::SourceLocation;
 use super::{DownloadError, DownloadStatus, ObjectFileSource, ObjectFileSourceURI};
 use crate::sources::{FileType, S3SourceConfig, S3SourceKey};
 use crate::types::ObjectId;
 
 type ClientCache = lru::LruCache<Arc<S3SourceKey>, Arc<rusoto_s3::S3Client>>;
 
+/// Maximum number of cached S3 clients.
+///
+/// This number defines the size of the internal cache for S3 clients and should be higher than
+/// expected concurrency across S3 buckets. If this number is too low, the downloader will
+/// re-authenticate between every request.
+///
+/// TODO(ja):
```

Comment:
unrelated, but what is the TODO here?

Reply:
To add the missing metrics. We had them for S3 but not for GCS.